Questions & Authorization

dabao40 opened the pull request:
I know, this might not be the appropriate place to ask such questions and to apply for such an authorization.  
I am a fan of Among Us as well as most of its mods, so it deeply depressed me when I heard that the author of TheOtherRolesGMH, haoming's team broke up. I have tried creating a mod branch out of the original mod TheOtherRolesGMH, but some of the files are missing, even if I have used the decompiler to decompile all its resources. Therefore, I will try to build a mod according and depending on your mod. However, when building this mod, I confronted some problems.  
![image](https://github.com/yukinogatari/TheOtherRoles-GM/assets/73883362/174abfc8-5509-4762-ac24-75d013720532)
Though these words were written in Chinese, it is easily seen that some of the variables such as "GameOptions", "SaveManager" etc. are not declared.  
I am willing to ask you this question and solve this problem.  
Besides, I will apply an authorization to you to see whether I can continue my build on this mod branch or not.  
  
Sincerely Yours,  
Imp11(dabao40) from China.

tomarai committed:
Add Madmate

The Madmate is a member of team Impostors, which has no special abilities.
The Madmate looks like a Crewmate to both other Crewmates and Impostors.
The Impostors do not know who is the Madmate and the Madmate does not know who are the Impostors.

When checking the win condition of the team Impostors, the Madmate is counted as a Crewmate.
The Madmate cannt fix comm and lightout sabotage.

tomarai committed:
Update README

tomarai committed:
Update README

gendelo3 committed:
Show an orange flash to the killer of the bait, can be turned of in settings

tomarai committed:
Add a custom option for Madmate to be able to fix comm sabotage

tomarai committed:
Update minor version number

tomarai committed:
Merge pull request #2 from tomarai/dev

Add a capability to Madmate to fix comm sabotage

tomarai committed:
Update README.md

tomarai committed:
Update README.md

tomarai committed:
Update README.md

tomarai committed:
Update README.md

tomarai committed:
Update README.md

yukinogatari committed:
A ton of changes (see the readme)

yukinogatari committed:
Update README.md

yukinogatari committed:
Update README.md

yukinogatari committed:
Update README.md

yukinogatari committed:
Update README.md

yukinogatari committed:
Disable admin/camera/vitals/vents options.

When blocking certain roles from fixing sabotages, they now can't interact with the device in the first place, making it somewhat less confusing for the player.

yukinogatari committed:
some more fixes in prep for some test games

* reworked the custom use button code entirely. now uses entirely separate UseButton objects instead of modifying the shown one, making it more flexible and less prone to error
* when use of a panel is blocked for any reason (madmate trying to fix lights, for example), the mod now displays a "blocked" message on the use button, preventing them from interacting with it in any way. this should make things a little more obvious for players
* added graphical buttons for the GM's zoom in/out ability
* made all players' positions visible on the GM's map
* disabled shadows for the GM when they're alive

yukinogatari committed:
test game fixes

* players' bodies are no longer randomly shuffled between games
* fixed morphing/camoflauger bugs introduced by my changes
* fixed the GM's zoom fucking up when you enter a meeting
* corpses show up on the GM's map too
* reworked the options display in the lobby to only show the roles that are enabled
* added translations for the detective/medic reports

yukinogatari committed:
final fixes for v2.9.0, probably

* show text on the vampire kill button, because the blank skull is really unnerving
* add support for translatable unit markers on custom options
* add support for translating images
* properly display disconnected players on the results screen
* more granular sorting on the results screen
* display the results in a more legible table format
* properly display skins on arsonist/GM/bounty hunter icons
* fix for the vampire's kill target not being outlined

yukinogatari committed:
Update README.md

yukinogatari committed:
small bug fixes

* guesser was unable to shoot in meetings, oops
* added the opportunist role by libhalt
* we now support a second repository for custom hats

yukinogatari committed:
update readme

yukinogatari committed:
bug fix: results screen doesn't display properly if the game ends during a camouflage/morph

yukinogatari committed:
opportunist is a solo team

yukinogatari committed:
bug fixes

* increase max impostors to 15 (why? why not)
* fix the UI getting lost if you open/close the map while zoomed out
* disable the selection entirely when trying to click on an invalid target in a meeting (GM, self if self vote disabled, skip button if skip disabled)

yukinogatari committed:
all sorts of fixes/updates

* sheriff option: misfire kills target
* security guard's cameras now display the room name properly
* reworked how arsonist icons display to be more consistent
* replace on/off special device restriction with time-based
* adjusted lovers quite a bit
  * option "can win with crew" replaced with "counts as separate team"
  * when off, behaves like old TOR properly (some fixes from the previous update left us inconsistent there)
* fix for airship showing up as dleks on the options menu
* expand impostor range to 0-15
* more granular info on the results screen, such as lovers suicide or arsonist torched
* hide number of tasks completed during comms sabotage
* add a pre-build step to the project to automatically compile our translation file into json

Eisbison committed:
Version 2.9.0

yukinogatari committed:
more fixes/new features

* try to compartmentalize what we consider as a "neutral" role so it's easier to work with them as a whole
* new option: randomize camouflager colors
* new option: guesser only show available roles
* add an overlay to display current settings/role summary in-game (currently only in JP)
* display a black screen briefly in the period between the report animation and the meeting starting
* allow for testing external hat repos through a config option
* change device restriction unit of time from minutes to seconds
* fix a bug in the hat system that prevented us from using multiple repos simultaneously
* updates to the results screen: a sabotage victory shows all non-impostors as dead, and a sheriff misfire has its own label
* attempt to fix a bug where a kill happening *after* a meeting begins results in a corpse being left behind for some players
* bring the guesser UI forward so it's not fogged up by the meeting phone screen
* don't highlight potential targets while inside a vent
* fix issues with the snitch arrows showing the wrong color sometimes
* opportunist is now handled as a neutral role instead of a crew role. as a result of this change, trying to shift into the opportunist (and the madmate) results in the shifter's death
* an erased neutral role no longer results in the player having to do tasks
* hide an arrow for a task completed during comms (original game bug)

dwd0tcom committed:
fix typo

Eisbison committed:
Version 2.9.1

Eisbison committed:
Merge branch 'main' of https://github.com/Eisbison/TheOtherRoles

yukinogatari committed:
fixes from last night's test play

* turn "hide settings" into a general option rather than a GM-specific option
* fix for some items not getting translated properly
* move opportunist/madmate up the list a bit to reflect their teams properly
* fix for the game not letting you start online if you have 0 impostors set
* don't let players see the info overlay if settings are hidden
* use the built-in mod stamp instead of a custom one because we've added so many buttons it's cumbersome to try and make it always visible
* don't run the end-game check during the intro
* crew can't win if there are no crewmates
* fix for the fix for deleting leftover bodies after a meeting
* opportunist's name wasn't showing up as green
* properly clear the misfired players list at the end of the game

yukinogatari committed:
Merge branch 'main' of https://github.com/Eisbison/TheOtherRoles into upstream-2.9.X

# Conflicts:
#	.gitignore
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptions.cs
#	TheOtherRoles/Helpers.cs
#	TheOtherRoles/Patches/CredentialsPatch.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/Patches/IntroPatch.cs
#	TheOtherRoles/Patches/MeetingPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/UpdatePatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
adjustments to the new upstream 2.9.x code that make it compatible with GM edition (mostly translation related)

yukinogatari committed:
few fixes to the 2.9.X updates

* translated the updater text
* fix for the game not ending if someone disconnects in the middle
* add a list of what counts as a "light" and a "dark" color to the in-game overlay
* updated README.md

yukinogatari committed:
updated readme

yukinogatari committed:
fix for morphing/camoflauger ending at the end of their timer

Eisbison committed:
v2.9.2

yukinogatari committed:
update README.md

yukinogatari committed:
new stuff

* fixes for some potential arsonist performance issues (don't check doused status every frame)
* mini: chance to be impostor
* eraser: cooldown penalty
* general: allow duplicate names
* general: randomize everyone's colors (not implemented)

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	README.md
#	TheOtherRoles/Main.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
fixes

Eisbison committed:
v3.0.0

gendelo3 committed:
Merge branch 'Eisbison:main' into bait_show_kill_flash

Eisbison committed:
v3.1.0

Eisbison committed:
v3.1.0

Eisbison committed:
v3.1.1

Eisbison committed:
Fix typo

Eisbison committed:
v3.1.2

gendelo3 committed:
Merge branch 'main' of https://github.com/Eisbison/TheOtherRoles into Eisbison-main

gendelo3 committed:
Merge branch 'Eisbison-main' into bait_show_kill_flash

dwd0tcom committed:
Fix typo

AcesFullOfKings committed:
Remove gendering from role descriptions

Girls play too

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/MapOptions.cs
#	TheOtherRoles/Objects/CustomButton.cs
#	TheOtherRoles/Patches/CredentialsPatch.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/Patches/IntroPatch.cs
#	TheOtherRoles/Patches/MeetingPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RegionMenuPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/UpdatePatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/TasksHandler.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
fixes to bring things in line with gm edition

yukinogatari committed:
fixes

* madmate can't enter vent
* madmate and jester can't sabotage
* modded colors not properly translated
* slight bug in mayor vote counter

yukinogatari committed:
Update README.md

Eisbison committed:
v3.2.0

Eisbison committed:
Merge branch 'main' of https://github.com/Eisbison/TheOtherRoles

dwd0tcom committed:
Hotfix for 3.2.0

dwd0tcom committed:
Merge pull request #247 from Eisbison/dev

Hotfix for 3.2.0

AcesFullOfKings committed:
Merge branch 'Eisbison:main' into patch-1

dwd0tcom committed:
Merge pull request #246 from theonefoster/patch-1

Remove gendering from role descriptions

Eisbison committed:
v3.2.2

Mallaris committed:
Update README.md

dwd0tcom committed:
Update README.md

yukinogatari committed:
fixed a bug with lovers not showing the description properly

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Helpers.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/Modules/CustomHats.cs
#	TheOtherRoles/Modules/CustomOptions.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/Patches/IntroPatch.cs
#	TheOtherRoles/Patches/MeetingPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/ShipStatusPatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/TasksHandler.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
Merge remote-tracking branch 'origin/gm-main' into gm-main

# Conflicts:
#	README.md

gendelo3 committed:
Merge branch 'Eisbison:main' into bait_show_kill_flash

yukinogatari committed:
bringing everything up to GM edition features

Eisbison committed:
v3.2.3

Developer626 committed:
Updated README.md file

Changed some of the wording of the Pursuer role to fix the grammar.

gendelo3 committed:
Merge branch 'Eisbison:main' into bait_show_kill_flash

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/Patches/ExileControllerPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

amsryq committed:
Revamped ClientOptionsPatch

Eisbison committed:
Merge pull request #252 from amsyarasyiq/main

Revamp ClientOptionsPatch

Eisbison committed:
Merge pull request #222 from gendelo3/bait_show_kill_flash

Show an orange flash to the killer of the bait, can be turned off in settings.

Eisbison committed:
v3.2.4

yukinogatari committed:
better way to handle end-of-game statuses

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Patches/ClientOptionsPatch.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
adding translations for the new update

yukinogatari committed:
fix readme

yukinogatari committed:
a few fixes

* allow hiding nameplates on an individual basis
* a failed shifter is marked as a suicide now
* fix for the lawyer stealing a win when they were ejected
* fix for players sometimes having their clothes/names swapped around
* hide the kill button on the meeting screen
* fix for non-vamp/warlock kills sometimes happening without a warp

yukinogatari committed:
updated README

yukinogatari committed:
fix for morphling not changing its name properly

yukinogatari committed:
minor bugfixes

* pets follow the player after they die
* visors show up below hats instead of above

yukinogatari committed:
bugfixes

* GM kill button wasn't being enabled
* player names weren't showing up on arsonist/gm/bounty hunter icons

yukinogatari committed:
some fixes

* add a way to prevent lovers being assigned to roles that are required to have tasks to function (snitch)
* properly hide nameplates on the report/dead players list screen
* fix the countdown not showing up for non-host players
* make the final role summary even more flexible
* add a "default" parameter to the string translation system

yukinogatari committed:
fix conditions for the lovers fix

now if we fail to assign a role bc of lovers conflicts, we retry 10 times and then remove that role from the pool

yukinogatari committed:
update version

yukinogatari committed:
build for the latest among us update

dwd0tcom committed:
bump version

dwd0tcom committed:
Merge pull request #263 from Eisbison/v-3-3-0

bump version

Mallaris committed:
Update README.md

dwd0tcom committed:
Merge pull request #250 from Developer626/patch-1

Updated README.md

tomarai committed:
Remove .DS_Store in Images

This commit

* Remove .DS_Store in Images
* Restore .DS_Store in .gitignore

tomarai committed:
Fix Evil Guesser cannot guess

Add null check to avoid for NullReferenceException in
`Guesser.remainingShots`.

Fix #264

dwd0tcom committed:
Merge pull request #265 from tomarai/remove-DS_Store

Remove .DS_Store in Images

dwd0tcom committed:
Merge pull request #266 from tomarai/fix-evilguesser-guess

Fix Evil Guesser cannot guess

dwd0tcom committed:
Version 3.3.1

dwd0tcom committed:
udpate README

dwd0tcom committed:
Merge pull request #268 from Eisbison/v3-3-1

Version 3.3.1

dwd0tcom committed:
version 3.3.2

dwd0tcom committed:
Merge pull request #269 from Eisbison/v3-3-2

version 3.3.2

yukinogatari committed:
fix for being unable to start games in the latest version

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/Modules/CustomHats.cs
#	TheOtherRoles/Modules/CustomOptions.cs
#	TheOtherRoles/Objects/CustomButton.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/Patches/MeetingPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
groundwork for multi-lovers, evil swapper, and twisted shifter

yukinogatari committed:
groundwork for corrupted shifter/ninja

moved our customized roles to a new file so it's easier to merge

dwd0tcom committed:
Version 3.3.3

dwd0tcom committed:
Merge branch 'main' into v-3-3-3

dwd0tcom committed:
Merge pull request #274 from Eisbison/v-3-3-3

Version 3.3.3

yukinogatari committed:
bug fixes

* adjust the timing on morphing/camo reset on meeting start
* handle pets properly with morphing/camo
* hide pet images on bottom-left player icons

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-main

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/UpdatePatch.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
fixing v3.3.3 for GM edition

yukinogatari committed:
evil swapper's name wasn't showing up red like it should have

yukinogatari committed:
some fixes

* new ninja options: can vent, can be targeted while stealthed
* new interface for pluralizing roles, implemented madmate and opportunist
* hide vent button for roles that can't vent

yukinogatari committed:
fixes weren't actually fixed orz

dwd0tcom committed:
Update issue templates

dwd0tcom committed:
Update issue templates

yukinogatari committed:
probably the final update for v3.3.0

* changes across the codebase to prepare for pluralizing all roles
* move the ninja button code into the ninja file
* better divide the shared per-instance code and per-role code
* fix a bug where a lover trying to shift their partner resulted in a one-person couple
* pluralize sheriff
* don't show the "lovers" role name anywhere, only the colored heart
  this makes the UI a bit cleaner and easier to read

yukinogatari committed:
don't highlight everyone in the sheriff's color

yukinogatari committed:
update readme, add some missing text

yukinogatari committed:
update readme

yukinogatari committed:
make roles a little more durable to disconnects

yukinogatari committed:
bugfixes, v3.3.1

* only show lover hearts in meetings during discussion/voting time
* properly make adaptive skins transparent
* display the current shifter's target on the meeting menu

yukinogatari committed:
bug fixes

* chain-shifter properly inherits arsonist, vulture, and sheriff status
* default ninja stealth duration reduced to 15s
* smooth out rendering of our help overlay/pre-meeting black screen
* show shifter target during meeting
* clean up ninja stealth code

Hageming committed:
Add madScientist

yukinogatari committed:
bug fixes, v3.3.2

* smarter handling lovers icon assignment
* sheriff options were bungled up

yukinogatari committed:
update version number

yukinogatari committed:
bug fixes

* eraseRole function wasn't working properly
* let GM kill/revive players during meetings

yukinogatari committed:
Merge pull request #2 from haoming37/madscientist_work

マッドサイエンティスト追加

yukinogatari committed:
groundwork for the mad scientist (by haoming37)

* rename to plague doctor ("mad" and "scientist" both overlap with other role names)
* add a few options (cooldown, multiple infections, post-meeting immunity time)
* only update the infection progress display after meetings, so the plague doctor can't tell when someone dies
* translation stuff (we're still missing some text)

yukinogatari committed:
was not handling FixedUpdate for roles even remotely correctly

yukinogatari committed:
only run the lovers ondeath function on actual lovers

yukinogatari committed:
pest doctor image

yukinogatari committed:
implement neko-kabocha

Hageming committed:
Add SerialKiller

yukinogatari committed:
bug fixes/changes

* reset ninja's opacity if they suddenly lose their role while invisible
* display the infection progress on the final screen

Hageming committed:
感染状況テキスト表示のnull参照修正

yukinogatari committed:
more flexible sidekick promotion & hide the arrows for the airship upload when completed during comms

yukinogatari committed:
Merge branch 'gm-main' into gm-dev

yukinogatari committed:
Merge pull request #4 from haoming37/plaguedoctor_fix

感染状況テキスト表示のnull参照修正

yukinogatari committed:
fix the exile code

yukinogatari committed:
Merge branch 'gm-main' into gm-dev

yukinogatari committed:
watcher? tetrachrome? idk still undecided on the name

yukinogatari committed:
lighter

* multi-lighter
* lighter can see invisible ninja (setting)
* smarter handling of lovers when they die

yukinogatari committed:
fix madmate win condition

yukinogatari committed:
update readme

yukinogatari committed:
properly erase plural roles and make it easier to not have this happen again

yukinogatari committed:
Merge pull request #3 from haoming37/serialkiller_work

シリアルキラー(misimo)追加

yukinogatari committed:
finally fix jackal/sidekick code

yukinogatari committed:
Merge branch 'gm-main' of https://github.com/yukinogatari/TheOtherRoles-GM into gm-main

yukinogatari committed:
serial killer

* add option to not reset timer after meeting
* ensure suicide timer is always longer than kill cool
* some fixes so you can have multiple of them

Hageming committed:
Fix stringsToJson, Force newline to \n

yukinogatari committed:
Merge pull request #6 from haoming37/fix_newline

Windowsでビルドした際にstringData.jsonの改行コードがCRLFになってしまう

Hageming committed:
Add Fox

yukinogatari committed:
don't show swap animation if swap didn't happen in time

Hageming committed:
Add FortuneTeller

dwd0tcom committed:
Version 3.4.0

dwd0tcom committed:
Merge branch 'main' into v-3-4-0

dwd0tcom committed:
Merge pull request #287 from Eisbison/v-3-4-0

Version 3.4.0

dwd0tcom committed:
Update README.md

dwd0tcom committed:
Update README.md

yukinogatari committed:
relocate post-meeting code so it runs at the correct time on airship

yukinogatari committed:
add a color gradation to the plague doctor infection progress %

yukinogatari committed:
strings

yukinogatari committed:
Merge branch 'gm-main' into gm-dev

# Conflicts:
#	Strings.xlsx
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/ShipStatusPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/Resources/stringData.json
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/Roles/ButtonsGM.cs
#	TheOtherRoles/Roles/CustomRolesGM.cs
#	TheOtherRoles/Roles/Lighter.cs
#	TheOtherRoles/Roles/Ninja.cs
#	TheOtherRoles/Roles/Role.cs
#	TheOtherRoles/Roles/Sheriff.cs

yukinogatari committed:
add serial killer button image

dwd0tcom committed:
Remove 3.3.4 from downloads

there was never a v3.3.4

dwd0tcom committed:
Version 3.4.1

dwd0tcom committed:
Merge branch 'main' into v-3-4-1

dwd0tcom committed:
Merge pull request #292 from Eisbison/v-3-4-1

Version 3.4.1

dwd0tcom committed:
Version 3.4.2

dwd0tcom committed:
Merge branch 'main' into v-3-4-2

dwd0tcom committed:
Merge pull request #293 from Eisbison/v-3-4-2

Version 3.4.2

dwd0tcom committed:
Update README.md

Hageming committed:
死亡している背徳者が再度死ぬバグを修正、マッドメイト死亡時にタスクカウントが正常に行えていなかったバグを修正

Hageming committed:
狐本人以外の狐のタスク数の分母が正常に表示されていなかったバグの修正

Hageming committed:
Rvert 狐本人以外の狐のタスク数の分母が正常に表示されていなかったバグの修正

yukinogatari committed:
Merge branch 'gm-main' into gm-tonsen

yukinogatari committed:
Merge pull request #5 from haoming37/fox_work

妖狐追加

yukinogatari committed:
2022.2.5 version

* add support for multiple loc xlsx files to make work on multiple branches easier
* change the version handshake to support #s greater than 255
* simplify handling of multi-team roles (either they're distributed equally or not)
* rename tetrachrome -> watcher

yukinogatari committed:
Merge remote-tracking branch 'origin/gm-dev' into gm-tonsen

# Conflicts:
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Helpers.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/ShipStatusPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/Roles/Role.cs

yukinogatari committed:
fixes for fox/immoralist/fortune teller

dwd0tcom committed:
Version 3.4.3

dwd0tcom committed:
Merge branch 'main' into v-3-4-3

dwd0tcom committed:
Merge pull request #300 from Eisbison/v-3-4-3

Version 3.4.3

dwd0tcom committed:
Update README.md

dwd0tcom committed:
Update README.md

yukinogatari committed:
fortune teller file formatting lol

yukinogatari committed:
fixes in prep for 3.5.0 release

* handle the situation in which everyone is dead at the end of the game
* plague doctor victims are now treated as dead at the end of the game
* dead ninja's clothes now properly go transparent
* fix for unnecessary text showing up around buttons when playing in english

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-tonsen

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Helpers.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/MapOptions.cs
#	TheOtherRoles/Modules/ChatCommands.cs
#	TheOtherRoles/Objects/CustomButton.cs
#	TheOtherRoles/Patches/ClientOptionsPatch.cs
#	TheOtherRoles/Patches/CredentialsPatch.cs
#	TheOtherRoles/Patches/IntroPatch.cs
#	TheOtherRoles/Patches/MeetingPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/ShipStatusPatch.cs
#	TheOtherRoles/Patches/UpdatePatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
Merge remote-tracking branch 'tor-eisbison/main' into gm-tonsen

# Conflicts:
#	README.md
#	TheOtherRoles/Buttons.cs
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Helpers.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/MapOptions.cs
#	TheOtherRoles/Modules/ChatCommands.cs
#	TheOtherRoles/Objects/CustomButton.cs
#	TheOtherRoles/Patches/ClientOptionsPatch.cs
#	TheOtherRoles/Patches/CredentialsPatch.cs
#	TheOtherRoles/Patches/IntroPatch.cs
#	TheOtherRoles/Patches/MeetingPatch.cs
#	TheOtherRoles/Patches/PlayerControlPatch.cs
#	TheOtherRoles/Patches/RoleAssignmentPatch.cs
#	TheOtherRoles/Patches/ShipStatusPatch.cs
#	TheOtherRoles/Patches/UpdatePatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/RoleInfo.cs
#	TheOtherRoles/TheOtherRoles.cs
#	TheOtherRoles/TheOtherRoles.csproj

Hageming committed:
背徳者→妖狐の矢印が表示されないことがある不具合の修正

yukinogatari committed:
final fixes for 3.5.0, probably?

* translations for the new options
* arsonist -> can't be lovers
* throw a warning if any option IDs overlap

yukinogatari committed:
Merge branch 'gm-tonsen' into gm-main

# Conflicts:
#	TheOtherRoles/CustomOptionHolder.cs
#	TheOtherRoles/Main.cs
#	TheOtherRoles/Patches/ShipStatusPatch.cs
#	TheOtherRoles/Patches/UsablesPatch.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
fix a couple merge errors

yukinogatari committed:
Merge pull request #8 from haoming37/fix_bug

背徳者→妖狐の矢印が表示されないことがある不具合の修正

yukinogatari committed:
Merge remote-tracking branch 'origin/gm-dev' into gm-tonsen

yukinogatari committed:
Update README.md

yukinogatari committed:
変更点

・ホスト以外でスワッパーのエフェクトが出ないバグを修正
・「ラバーズチャット」有効時、会議画面がバグってた
・誰もターゲットしてない状態で能力ボタンを押したらクールだけ消費しちゃうバグを修正
・「淡い・濃いを表示する」のデフォルト設定をオフに変更
・翻訳漏れがあった

yukinogatari committed:
変更点

・ホスト以外でスワッパーのエフェクトが出ないバグを修正
・「ラバーズチャット」有効時、会議画面がバグってた
・誰もターゲットしてない状態で能力ボタンを押したらクールだけ消費しちゃうバグを修正
・「淡い・濃いを表示する」のデフォルト設定をオフに変更
・翻訳漏れがあった

# Conflicts:
#	TheOtherRoles/Main.cs
#	TheOtherRoles/Patches/EndGamePatch.cs
#	TheOtherRoles/RPC.cs

yukinogatari committed:
some fox-related fixes

yukinogatari committed:
Merge remote-tracking branch 'origin/gm-main' into gm-main

yukinogatari committed:
properly end the game on v2022.2.8s

yukinogatari committed:
Merge branch 'gm-main' into gm-tonsen

yukinogatari committed:
fix fox tasks

yukinogatari committed:
handle task generation better, and make it more future proof

yukinogatari committed:
few minor fixes

* chain-shifter couldn't properly shift into neko-kabocha
* chain-shifter now becomes the lawyer's client when shifting with them
* lawyer can now properly see their client's role when the setting is enabled
* bait can't be lovers anymore

yukinogatari committed:
Merge branch 'gm-main' into gm-tonsen

# Conflicts:
#	TheOtherRoles/Main.cs
#	TheOtherRoles/RPC.cs
#	TheOtherRoles/TheOtherRoles.csproj

yukinogatari committed:
3.5.3

* add missing fox button
* better support for games with roles disabled
* bring over the TOR patch to connect to official servers properly
* make a custom option just for tasks

yukinogatari committed:
update for tonsen's first march edition

* madmate is now an attribute, allowing it to double up with other roles
* madmate fanatic mode
* the GM is not a crewmate
* calculate madmate/fox tasks in the same function

yukinogatari committed:
Merge branch 'gm-tonsen' into gm-main

yukinogatari committed:
final fixes before 3.6.0

* implemented simplified chinese translation (partial)
* new option to choose whether to refund votes when a player is killed during the meeting
* new option to hide task arrows
* remove no-longer-relevant fix for missing airship admin button
* don't show plague doctor progress when a plague doctor doesn't exist
* fix mad bait so it properly selects an existing player to report
* remove fortune teller and fox from lovers candidates if lovers can't do tasks
* prevent task-dependent roles from becoming madmate (snitch, fortune teller)
* ensure arrows exist before destroying them in fortune teller/fox/immoralist code
* properly handle neko-kabocha revenge kill when shot by guesser
* clean up some meeting code so bugs like above are less likely to reoccur

yukinogatari committed:
changed version # to 3.5.4

* implemented the rest of the Simplified Chinese translation
* fixed a bug with the fox's win condition
* fixed a bug where a fox disconnecting would allow them to see all players' roles in future games

yukinogatari committed:
partially update to v2022.3.29s

* add sprinter
* disable morphling until we can figure out how to make it work again
* pet prefab went missing, so we have to figure out a new way to spawn other people's pets for morphing
* HatBehaviour renamed to HatData, largely works the same otherwise (altshader seems to have gone missing tho)
* Dleks was removed from the game, so remove the ability to select it
* the role intro sequence was changed to a coroutine, making it harder to modify (temp solution still shows the original "your role is Crewmate" for a frame before swapping in the correct role)
* update nameplate patch, bc the first one in the list is no longer the blank nameplate

yukinogatari committed:
akujo basically complete

* merge in TOR's changes
* hats fixed
* fix RoleInfo so it loads after the options are loaded
* fix pets/morphing
* make it easier to set meeting info text for any role
* unify condition for showing extra info during meetings
* reorganize roleinfo so guesser roles show up in more logical order
*

yukinogatari committed:
Merge branch 'gm-tonsen' into gm-main

yukinogatari committed:
update readme

yukinogatari committed:
few fixes from 3.5.5

* fix a bug that prevented jackal/fox/eraser from being able to create sidekicks/erase roles
* implemented sc translation for sprinter
* pass information about what role we're changing into when erasing old role info

yukinogatari committed:
some changes I never committed? idk, but making it public anyway

yukinogatari committed:
add akujo text

yukinogatari commented:
This mod is based on a version of the game nearly a year and a half old. A lot is going to have changed in that time, and it will probably take a ton of work to make it compatible with the latest version. You are free to use the code in this repository however you want, but I don't play anymore, so I won't be able to help with fixing any issues you run into.

dabao40 commented:
Thank you very much! I have added most of the roles from The Other Roles GM into my own mod and have also fixed some bugs. I sincerely appreciate your work and I thank you for your understanding. Thank you again.
